Prox/Prox/Widgets/ExpandingCardView.swift

```diff
@@ -0,0 +1,135 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+// NOTE: [mcomella] It'd be cool to turn this into a library, which is why it is structured
+// as one. Please keep it clean, if possible! :)
+
+// TODO:
+// - TODOs in file
+// - Option to animate layout changes?
+// - Verify access control
+// - Break out into library (test using elsewhere, samples, readme w/ public methods;
+//   show how we can use constraint(equalTo:) and constraint(lessThanOrEqualTo:)
+
+/// A floating card view which will expand to display its full content without scrolling.
+/// If the height is constrained, the content will automatically be scrollable.
+///
+/// The content view can be assigned with `expandingCardView.contentView = ...`. To correctly
+/// expand, the content view must have a defined height: e.g. a UIStackView, a view with
+/// a heightAnchor, or a view with intrinsicContentSize (like UILabel).
+public class ExpandingCardView: UIView {
+
+    // MARK: public API
+
+    public override var backgroundColor: UIColor? {
+        get { return scrollView.backgroundColor }
+        set { scrollView.backgroundColor = newValue }
+    }
+    public var cornerRadius: CGFloat {
+        get { return scrollView.layer.cornerRadius }
+        set { scrollView.layer.cornerRadius = newValue }
+    }
+
+    public var shadowColor: CGColor? {
+        get { return layer.shadowColor }
+        set { layer.shadowColor = newValue }
+    }
+    public var shadowOpacity: Float {
+        get { return layer.shadowOpacity }
+        set { layer.shadowOpacity = newValue }
+    }
+    public var shadowRadius: CGFloat {
+        get { return layer.shadowRadius }
+        set { layer.shadowRadius = newValue }
+    }
+    public var shadowOffset: CGSize {
```

Comment:
Any specific reason you're exposing these `layer` properties as properties on the view? If I were using this class, I would expect to do `expandingCardView.layer.shadowColor = ...` like any other `UIView`.